src/util/dev/server.ts

```diff
@@ -1102,6 +1102,11 @@ export default class DevServer {
         );
         return;
       }
+
+      if (status === 404) {
```

Comment:
This is not correct, as it prevents a custom 404 page in your deployment. This if-block was removed in #2689.

Reply:
@TooTallNate I removed it, but then this route will not work:

```
{
      status: 404,
      src: '/api(\\/.*)$',
}
```

but it does in production